### PR TITLE
Simplify DoFCellAccessorImplementation::update_cell_dof_indices_cache()

### PR DESCRIPTION
**Pull request by peterrum**
This PR simplifies the function `DoFCellAccessorImplementation::Implementation::update_cell_dof_indices_cache()` by:
* delegating the task of getting the indices to `dealii::DoFAccessor::get_dof_indices()` and
* getting the pointer to the cache from `DoFLevel`.

In the spirit of #9644, #9760, and #9791.

~All tests `ctest -j30 -R "hp|dofs|matrix_free"` have passed. Currently, I am running the complete test suite with this configuration https://github.com/dealii/dealii/pull/9760#issuecomment-605505157.~  All tests with the configuration https://github.com/dealii/dealii/pull/9760#issuecomment-605505157 have passed.

### CODE REVIEW

File: include/deal.II/dofs/dof_accessor.templates.h [hunk 1]
```diff
@@ -2746,197 +2746,12 @@ namespace internal
     struct Implementation
     {
       /**
-       * Implement the updating of the cache. Currently not implemented for
-       * hp::DoFHandler objects.
```
Comment (peterrum) [inline include/deal.II/dofs/dof_accessor.templates.h:2750]: @bangerth A lot has happened since this statement was written ;)

**Comment by masterleinad** [comment]
@bangerth /rebuild has to be after setting the `ready to test` label.

**Comment by kronbichler** [comment]
@peterrum it seems we need a rebase here to re-try with travis

**Comment by kronbichler** [comment]
If I remember correctly, we had the github-CI OSX builds passing before the rebase, so I'm taking the risk now and merge as the new failures are unrelated.

**Pull request merged**